react/src/components/ImageList.tsx

```diff
@@ -0,0 +1,369 @@
+import Flex from '../components/Flex';
+import { getImageFullName } from '../helper';
+import { useBackendAIImageMetaData, useUpdatableState } from '../hooks';
+import ImageInstallModal from './ImageInstallModal';
+import { ConstraintTags } from './ImageTags';
+import ManageAppsModal from './ManageAppsModal';
+import ManageImageResourceLimitModal from './ManageImageResourceLimitModal';
+import ResourceNumber from './ResourceNumber';
+import {
+  ImageListQuery,
+  ImageListQuery$data,
+} from './__generated__/ImageListQuery.graphql';
+import {
+  AppstoreOutlined,
+  SettingOutlined,
+  VerticalAlignBottomOutlined,
+} from '@ant-design/icons';
+import { Button, Table, Tag, theme } from 'antd';
+import { AnyObject } from 'antd/es/_util/type';
+import { ColumnsType } from 'antd/es/table';
+import { ColumnType } from 'antd/lib/table';
+import graphql from 'babel-plugin-relay/macro';
+import { Key, useMemo, useState, useTransition } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLazyLoadQuery } from 'react-relay';
+
+export type EnvironmentImage = NonNullable<
+  ImageListQuery$data['images']
+>[number];
+
+export default function ImageList({ style }: { style?: React.CSSProperties }) {
+  const { t } = useTranslation();
+  const [selectedRowKeys, setSelectedRowKeys] = useState<Key[]>([]);
+  const [
+    ,
+    {
+      getNamespace,
+      getBaseVersion,
+      getImageLang,
+      getBaseImages,
+      getConstraints,
+    },
+  ] = useBackendAIImageMetaData();
+  const { token } = theme.useToken();
+  const [managingApp, setManagingApp] = useState<EnvironmentImage | null>(null);
+  const [managingResourceLimit, setManagingResourceLimit] =
+    useState<EnvironmentImage | null>(null);
+  const [isOpenInstallModal, setIsOpenInstallModal] = useState<boolean>(false);
+  const [environmentFetchKey, updateEnvironmentFetchKey] =
+    useUpdatableState('initial-fetch');
+  const [, startRefetchTransition] = useTransition();
+  const [installingImages, setInstallingImages] = useState<string[]>([]);
+
+  const onSelectChange = (newSelectedRowKeys: Key[]) => {
+    setSelectedRowKeys(newSelectedRowKeys);
+  };
+
+  const rowSelection = {
+    selectedRowKeys,
+    onChange: onSelectChange,
+  };
+
+  const sortBasedOnInstalled = (a: EnvironmentImage, b: EnvironmentImage) => {
+    return a?.installed && !b?.installed
+      ? -1
+      : !a?.installed && b?.installed
+        ? 1
+        : 0;
+  };
+
```

Comment:
https://github.com/lablup/backend.ai-webui/blob/main/src/components/backend-ai-environment-list.ts#L461
lit code for reference